### ISSUE TITLE
FEATURE: Fallback when there is no internet connection

### ISSUE THREAD

**Issue by Rikiub** [GnoxNahte/obsidian-auto-embed#21]
It would be great if when there is no internet connection instead of showing this (From my Android device):

![Screenshot_20241223-234410_Obsidian](https://github.com/user-attachments/assets/058b9567-d79f-4e9c-b321-b37b5355f1b7)

Display a placeholder like this with text like **"Offline"** or **"No connection"**.

![Screenshot_20241223-234559_Obsidian](https://github.com/user-attachments/assets/98c3a842-43c9-4bee-807f-e55e7542514b)

**Comment by GnoxNahte** [comment]
Added it! Thanks for the suggestion. 

It should look like this. I don't have an Android but it should work.
![image](https://github.com/user-attachments/assets/8888c018-ae4d-491a-9c3c-21ea1cc2db42)

**Comment by Rikiub** [comment]
_Wow! It was really fast!_

Unfortunately it still shows the default Android error on my device, but should works on desktop, I guess.
I'll test it when I get home.

**Comment by GnoxNahte** [comment]
Hmm, I'm using this to check [navigator.onLine](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/onLine). It does say this in the docs 
> In Chrome and Safari, if the browser is not able to connect to a local area network (LAN) or a router, it is offline; all other conditions return true. So while you can assume that the browser is offline when it returns a false value, you cannot assume that a true value necessarily means that the browser can access the internet. You could be getting false positives, such as in cases where the computer is running a virtualization software that has virtual ethernet adapters that are always "connected." Therefore, if you really want to determine the online status of the browser, you should develop additional means for checking.

TLDR: might get some false positives - might return "there's internet" even when it is be offline. 

I'll try to implement a proper check but it'll take slightly longer. I'll do it whenever I have more free time. It might be hard for me to test as I don't have an Android though.

**Comment by Rikiub** [comment]
Yeah! Take your time!
For now I'm looking how to make this works.

I think make pings to websites like google.com should check if there is internet connection.
But I'm sure there are better ways.
